src/components/CarouselCard/index.js

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable no-script-url,max-len */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Provider } from './context';
+import './styles.css';
+import Indicators from './indicators';
+
+export default class CarouselCard extends Component {
+    constructor(props) {
+        super(props);
+        this.registerChild = this.registerChild.bind(this);
+        this.setActiveItem = this.setActiveItem.bind(this);
+        this.state = {
+            childs: [],
+            activeItem: undefined,
+            privateRegisterChild: this.registerChild,
+        };
+        this.cardPosition = { transform: 'translateX(-0%)' };
+    }
+
+    componentDidMount() {
+        this.startAnimation();
+    }
+
+    setActiveItem(id) {
+        const { childs } = this.state;
+        const selectedItemIndex = childs.findIndex(child => child.indicatorID === id);
+        this.cardPosition = { transform: `translateX(-${selectedItemIndex}00%)` };
+        this.setState({ activeItem: id });
+    }
+
+    startAnimation() {
+        const { scrollDuration, disableAutoScroll } = this.props;
+        if (!disableAutoScroll) {
+            return setTimeout(() => {
+                const { childs, activeItem } = this.state;
+                const selectedItemIndex = childs.findIndex(child => child.indicatorID === activeItem);
```

Comment:
here you can create a new function findSelectedItemIndex and put it in a new file to later test it. Above in setActiveItem method you can reuse this function too.